### PR TITLE
Branch search plan

### PR DESCRIPTION
**Pull request by a1021492980**
This PR contains file reading and writing for plans.
And it also includes search feature for plan.

### CODE REVIEW

File: src/main/java/commands/SearchCommand.java [hunk 1]
```diff
@@ -48,13 +67,31 @@ public void setUserAction(String userAction) throws InvalidCommandException {
...
+            if (isFirstMatch()) {
+                printHeadingMessage("plan");
+            }
+            break;
+        default:
+            logger.log(Level.WARNING, "User has entered an invalid search exercise command action.");
+            String className = this.getClass().getSimpleName();
+            throw new InvalidCommandException(className, InvalidCommandException.INVALID_ACTION_ERROR_MSG);
         }
+
```
Comment (alanlowzies) [inline src/main/java/commands/SearchCommand.java:87]: Perhaps this extra empty line can be removed?

**Review by alanlowzies — APPROVED** [review]
Great work with the plans file and the search function. I just have a super minor code quality suggestion for your consideration (just an extra empty line actually). I will do the schedule file writing and reading. Thank you

File: src/main/java/commands/SearchCommand.java [hunk 2]
```diff
@@ -87,16 +124,36 @@ public void searchExercise() {
         }
     }
 
-    public boolean isMatchExercise(String exercise) {
+    public boolean isMatch(String exercise) {
         return exercise.contains(userArguments);
```
Comment (Musfirahe0556596) [inline src/main/java/commands/SearchCommand.java:128]: I think the parameter name "exercise" should be changed as this method is more generalized now?

File: src/main/java/commands/SearchCommand.java [hunk 3]
```diff
@@ -87,16 +124,36 @@ public void searchExercise() {
...
     }
 
+    public void searchPlan() throws InvalidCommandException {
+        var planListToSearch = planList.getPlansDisplayList();
+        for (int i = 0; i < planListToSearch.size(); i++) {
+            String planToBeCompared = planListToSearch.get(i).toLowerCase();
+            var planHash = planList.getPlanFromKey(planToBeCompared);
+            var workouts = planHash.getWorkoutsInPlanList();
```
Comment (Musfirahe0556596) [inline src/main/java/commands/SearchCommand.java:136]: Just wondering, are these variables used anywhere?

**Review by Musfirahe0556596 — APPROVED** [review]
Done very well! :) Just some minor changes!

File: src/main/java/commands/SearchCommand.java (hunk 2 above)
Comment (a1021492980) [reply src/main/java/commands/SearchCommand.java:128]: yes, thank you for the reminder!

File: src/main/java/commands/SearchCommand.java (hunk 3 above)
Comment (a1021492980) [reply src/main/java/commands/SearchCommand.java:136]: I forget to remove, thank you!

File: src/main/java/commands/SearchCommand.java (hunk 1 above)
Comment (a1021492980) [reply src/main/java/commands/SearchCommand.java:87]: Yes, I didn't see that, thank you!

**Pull request merged**